### PR TITLE
qa_crowbarsetup: unify crowbar upgrade api requests

### PR DESCRIPTION
**Pull request by jdsn**
This is the 2nd try to get PR #1100  merged (with all needed gating runs).
PR #1100 was merged to early and was thus reverted by PR #1104 .

**PLEASE NOTE**

This PR may be merged with enough +1s, a successful 'normal' gating run **and** a failing 'upgrade' gating run with error **422** in an api request to the services endpoint (this failure is expected). The request to the 'start' endpoint however is not expected.
@MaximilianMeister is this still correct?

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4436,6 +4436,20 @@ function onadmin_is_crowbar_api_available()
     [[ $http_code =~ [23].. ]]
 }
 
+function crowbar_upgrade_api()
+{
+    local method=$1
+    local endpoint=$2
+    local curlopts=${3:-''}
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: this looks a bit strange:
```
11:19:13 +(qa_crowbarsetup.sh:4447) crowbar_upgrade_api(): local 'curlopts=-F file=@/tmp/backup-crowbar.tar.gz'
```
but somehow gets rendered correctly:
```
11:19:13 ++(qa_crowbarsetup.sh:4450) crowbar_upgrade_api(): curl -X POST -F file=@/tmp/backup-crowbar.tar.gz -s -o crowbar-upgrade-api-start.txt -w '%{http_code}' --digest -u crowbar:crowbar http://localhost:3000/installer/upgrade/start.json
```

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4451,11 +4465,7 @@ function onadmin_crowbarrestore()
             with_upgrade)
                 # restore after upgrade has different workflow (missing APIs) than
                 #   a restore from a backup of the same cloud release
-                local http_code=`curl -X POST -F "file=@/tmp/$btarball" -s -o upgrade-start.txt -w '%{http_code}' $crowbar_api/installer/upgrade/start.json`
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: the other thing/difference is we didn't use authentification here, and now we are (its also not necessarily needed, as we are already on the admin node), but that shouldn't make any difference

**Comment by jdsn** [comment]
the PR now includes the ideas from PR #1212  (1212 is obsolete with this)

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: with the new commits now the unneeded authentication is removed again - so it should be the same as before

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1765,35 +1766,16 @@ function onadmin_bootstrapcrowbar()
...
-        local http_code
         install_crowbar_init
-        # FIXME: when https://github.com/SUSE-Cloud/automation/pull/1105 is merged
-        http_code=`curl -s -X POST -H "Accept: application/vnd.crowbar.v2.0+json" --data "username=crowbar&password=crowbar" -o setup-database.txt -w '%{http_code}' $crowbar_init_api/database/new`
-        if ! [[ $http_code =~ [23].. ]] ; then
-            cat setup-database.txt
-            complain 36 "Could not setup PostgreSQL database"
-        fi
+        crowbar_api_request POST $crowbar_init_api /database/new \
+            '--data "username=crowbar&password=crowbar"'  "$crowbar_api_v2_header"
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: there is an unnecessary whitespace afaics

**Comment by MaximilianMeister** [comment]
> the PR now includes the ideas from PR #1212 (1212 is obsolete with this)

not fully. see https://github.com/SUSE-Cloud/automation/pull/1212#issuecomment-245594824
